src/VisualStudio/Core/Def/Implementation/LanguageService/AbstractPackage`2.cs

```diff
@@ -49,6 +50,9 @@ protected override async Task InitializeAsync(CancellationToken cancellationToke
             Assumes.Present(shell);
             Assumes.Present(solution);
 
+            var settingsEditorFactory = ComponentModel.GetService<SettingsEditorFactory>();
+            RegisterEditorFactory(settingsEditorFactory);
```

Comment:
So this means we're registering one per language? Which package is the pkgdef defining the factory against?

Reply:
> So this means we're registering one per language

You can re-register an editor factory as many times as you want. because its keyed off of the guid it will still only create one and have one registered. So worst case scenario we register it twice, if there are both C# and VB projects. But we get the advantage of only taking over editorconfig files if a package loads for a programming language we understand

Reply:
>  Which package is the pkgdef defining the factory against?

This is defined in Core/Def here:
https://github.com/dotnet/roslyn/blob/main/src/VisualStudio/Core/Def/PackageRegistration.pkgdef#L28-L37

And is using the RoslynPackage Id. Again, this has the effect of 
- only one editor factory ever exists (because there is only one pkdef entry for it and therefore it is associated with only one GUID)
- The editor factory only becomes "active" (i.e. registered) when a C# or Visual Basic project loads.

Reply:
My concern here is the package base type sometimes looks at the pkgdef entries or something else and might decide that since it's seeing an editor factory registered for another package it may not handle it right. But how is this helping? If our package isn't loaded, the shell should load RoslynPackage. This feels like we're missing a root cause here.

Reply:
> If our package isn't loaded, the shell should load RoslynPackage. This feels like we're missing a root cause here.

The C++ team does not want our editorfactory to load in a pure C++ project. Once this UI has been productized and the C++  has had an opportunity to plug in then it will be ok. Today the experience we want is to _not_ load

Reply:
I'm confused. The factory is either registered or not, at any point in time. How does moving the registration code help that?

Reply:
> The factory is either registered or not

right on opening a C++ project it will not have been registered. This conversation points out an issue though. If the someone were to load a C# project, close the solution, and then open a C++ project we would still take over the editor. In the case of an C++ project we would fail to populate options and just show a blank UI. I may back the registration code out and just keep the project null check.

Reply:
I don't exactly know how it works but there is interfaces like [IVsEditorFactoryChooser](https://docs.microsoft.com/en-us/dotnet/api/microsoft.visualstudio.shell.interop.ivseditorfactorychooser) that can be used to more dynamically choose editor factories. I'm not precisely sure how to use it but it looks like it's given an IVsHierarchy so we can use that to figure out if it's a C#/VB project or not.

(Independently I'm still worried we have a bug since the editor factory load should trigger the package load, so the fact your change did anything at all still seems wrong.)